src/inference/src/dev/core_impl_ie.cpp

```diff
@@ -0,0 +1,385 @@
+        } else {
+            // Temporary workaround until all plugins support caching of original model inputs
+            InferenceEngine::SetExeNetworkInfo(res._ptr, network.getFunction(), isNewAPI());
+        }
+    } else {
+        res = LoadNetworkImpl(network, plugin, parsed._config, nullptr, cacheContent, forceDisableCache);
+    }
+    return {res._ptr, res._so};
+}
+
+InferenceEngine::SoExecutableNetworkInternal ov::CoreImpl::LoadNetwork(
+    const std::string& modelPath,
+    const std::string& deviceName,
+    const std::map<std::string, std::string>& config,
+    const std::function<void(const InferenceEngine::CNNNetwork&)>& val) {
+    OV_ITT_SCOPE(FIRST_INFERENCE, ie::itt::domains::IE_LT, "Core::LoadNetwork::Path");
+    auto parsed = parseDeviceNameIntoConfig(deviceName, config);
+    auto plugin = GetCPPPluginByName(parsed._deviceName);
+    ov::SoPtr<InferenceEngine::IExecutableNetworkInternal> res;
+    auto conf = any_copy(parsed._config);
+    auto cacheManager =
+        coreConfig.get_cache_config_for_device(parsed._deviceName, device_supports_cache_dir(plugin), conf)
+            ._cacheManager;
+    auto cacheContent = CacheContent{cacheManager, modelPath};
+    if (cacheManager && device_supports_import_export(plugin)) {
+        bool loadedFromCache = false;
+        cacheContent.blobId = CalculateFileHash(modelPath, parsed._deviceName, plugin, conf);
+        auto lock = cacheGuard.getHashLock(cacheContent.blobId);
+        res = load_model_from_cache(cacheContent, plugin, conf, {}, loadedFromCache);
+        if (!loadedFromCache) {
+            auto cnnNetwork = ReadNetwork(modelPath, std::string());
+            if (val) {
+                val(cnnNetwork);
+            }
+            if (cnnNetwork.getFunction()) {
+                res = compile_model_impl(ov::legacy_convert::convert_model(cnnNetwork, isNewAPI()),
+                                         plugin,
+                                         conf,
+                                         {},
+                                         cacheContent);
+            } else {
+                res = LoadNetworkImpl(cnnNetwork, plugin, parsed._config, nullptr, cacheContent);
+            }
+        }
+    } else if (cacheManager) {
+        res = plugin.compile_model(modelPath, conf);
+    } else {
+        auto cnnNetwork = ReadNetwork(modelPath, std::string());
+        if (val) {
+            val(cnnNetwork);
+        }
+        if (cnnNetwork.getFunction()) {
+            res = compile_model_impl(ov::legacy_convert::convert_model(cnnNetwork, isNewAPI()),
+                                     plugin,
+                                     conf,
+                                     {},
+                                     cacheContent);
+        } else {
+            res = LoadNetworkImpl(cnnNetwork, plugin, parsed._config, nullptr, cacheContent);
+        }
+    }
```

Comment:
why don't we recall compile_model here?

Reply:
The same answer as before. The problem with IR v7